prow/cmd/build/controller_test.go

```diff
@@ -161,6 +182,62 @@ func (fl *fakeLimiter) NumRequeues(item interface{}) int {
 	return 0
 }
 
+type fakeNsLister struct {
+	pjs map[string]prowjobv1.ProwJob
+}
+
+func newFakeNsLister(pjs map[string]prowjobv1.ProwJob) *fakeNsLister {
+	return &fakeNsLister{
+		pjs: pjs,
+	}
+}
+
+func (f *fakeNsLister) List(selector labels.Selector) ([]*prowjobv1.ProwJob, error) {
+	result := []*prowjobv1.ProwJob{}
+	for _, pj := range f.pjs {
+		newPj := pj
+		result = append(result, &newPj)
+	}
+	return result, nil
+}
+
+func (f *fakeNsLister) Get(name string) (*prowjobv1.ProwJob, error) {
+	pj, found := f.pjs[name]
+	if !found {
+		return nil, fmt.Errorf("job %q not found", name)
+	}
+	return &pj, nil
+}
+
+type fakeLister struct {
+	allPjs map[string]map[string]prowjobv1.ProwJob
+}
+
+func newFakeLister(allPjs map[string]map[string]prowjobv1.ProwJob) *fakeLister {
```

Comment:
Why not use the fake clients from upstream?

Reply:
It seems that `Lister` does not have a fake implementation https://github.com/kubernetes/code-generator/tree/master/cmd/lister-gen/generators. Do you know any available?

Reply:
Hope this helps? 

https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/configobservation/apiserver/observe_apiserver_test.go#L62-L71

Reply:
Thanks! This was helpful. I switched to the cache.Indexer.